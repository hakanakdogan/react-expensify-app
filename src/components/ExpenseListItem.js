import React from "react";
import { Link } from "react-router-dom";


 const ExpenseListItem = ({ history, description, amount, createdAt, id, dispatch }) => (

    <div>
        <Link to={`/editexpense/${id}`}>
            <h3>{description}</h3>
        </Link>

        <ul>

            <li>Amount: {amount}</li>
            <li>Created At: {createdAt}</li>

        </ul>
        

        {
            // <Link to={`/editexpense/${id}`}>
            //     <button>
            //         Edit
            //     </button>
            // </Link>

            //Butonu sayfa değiştirmek için böyle kullanabiliyorum
        }


        <hr />
    </div>

);


export default ExpenseListItem;