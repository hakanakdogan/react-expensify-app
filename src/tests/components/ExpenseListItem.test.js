import React from "react";
import { shallow } from "enzyme";
import  ExpenseListItem  from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

const expense = expenses[0];


test("Should render ExpenseListItem with given expense", () => {
    const wrapper = shallow(<ExpenseListItem key={expense.id} {...expense} />);
    expect(wrapper).toMatchSnapshot();
});