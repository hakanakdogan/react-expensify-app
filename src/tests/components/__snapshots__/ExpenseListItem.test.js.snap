// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem with given expense 1`] = `
<div>
  <Link
    replace={false}
    to="/editexpense/1"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <ul>
    <li>
      Amount: 
      195
    </li>
    <li>
      Created At: 
      0
    </li>
  </ul>
  <hr />
</div>
`;
